### ISSUE TITLE
ArcGIS Server: Enterprise Geodatabases: IsManaged=false Ignored by Server Manager

### ISSUE THREAD

**Issue by ZianAtFirstWatch** [Esri/arcgis-powershell-dsc#349]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.2.0

### Affected Resource(s)

* ArcGIS_Server

### Configuration Files

```
{
    "AllNodes": [{
            "NodeName": "StoutESRI2",
            "DataStoreTypes": [
                "Relational"
            ],
            "Role": [
                "DataStore",
                "Server",
                "SQLServerClient"
            ]
        }
    ],
    "ConfigData": {
        "Version": "10.7.1",
        "ServerContext": "fw_arcgisserver",
        "PortalContext": "fw_arcgisportal",
        "ServerRole": "GeneralPurposeServer",
        "Credentials": {
            "ServiceAccount": {
                "Password": "bleep",
                "UserName": "datacenter\\esri_admin",
                "IsDomainAccount": true,
                "IsMSAAccount": false
            }
        },
        "SQLServerClient": [{
                "Name": "Microsoft ODBC Driver 17 for SQL Server",
                "InstallerPath": "C:\\PowerShell DSC Installers\\10.7.1\\msodbcsql_17.3.1.1_x64.msi",
                "ProductId": "175BC9FC-C256-4EFC-9D87-4A9AF3B47A12",
                "Arguments": "/L*V \"C:\\users\\fwla\\desktop\\odbc installation.log\" /quiet /qn IACCEPTMSODBCSQLLICENSETERMS=YES ADDLOCAL=ALL"
            }
        ],
        "Server": {
            "LicenseFilePath": "C:\\PowerShell DSC Installers\\ArcGISGISServerStandard_ArcGISServer_1100345.prvc",
            "Installer": {
                "Path": "C:\\PowerShell DSC Installers\\10.7.1\\ArcGIS_Server_Windows_1071_169677.exe",
                "InstallDir": "C:\\Program Files\\ArcGIS\\Server",
                "InstallDirPython": "C:\\Python27",
                "PatchesDir": "C:\\PowerShell DSC Installers\\AGS Updates"
            },
            "ServerDirectoriesRootLocation": "C:\\arcgisserver\\directories",
            "ConfigStoreLocation": "C:\\arcgisserver\\config-store",
            "PrimarySiteAdmin": {
                "UserName": "fwadmin2",
                "Password": "bloop"
            },
            "Databases": {
                "DatabaseServerHostName": "FW_ArcGIS_SQL_SitePrimary",
                "DatabaseName": "arcsde",
                "DatabaseType": "SQLServerDatabase",
                "DatabaseAdminUser": {
                    "UserName": "SQL_Apps",
                    "Password": "dotdotdot"
                },
                "DatabaseUser": {
                    "UserName": "SQL_Apps",
                    "Password": "dotdotdot"
                },
                "IsManaged": false,
                "EnableGeodatabase": true
            }
        },
        "DataStore": {
            "ContentDirectoryLocation": "C:\\arcgisdatastore",
            "EnableFailoverOnPrimaryStop": false,
            "Installer": {
                "Path": "C:\\PowerShell DSC Installers\\10.7.1\\ArcGIS_DataStore_Windows_1071_169689.exe",
                "InstallDir": "C:\\Program Files\\ArcGIS\\DataStore",
                "PatchesDir": "C:\\PowerShell DSC Installers\\Data Store Updates"
            }
        }
    }
}

```

### Expected Behavior

I expected to see something like the following screenshot in AGS Server Manager:
![Happy Screenshot](https://user-images.githubusercontent.com/32464722/141925210-a8fa2542-a594-4b9f-8fd1-9ea820487b70.png)

Though without the 2 different connection strings, as I didn't provide the PowerShell module with both logins.

### Actual Behavior

The "Managed database" checkbox was checked as shown below:
![Sad Screenshot](https://user-images.githubusercontent.com/32464722/141924999-56f7049e-a788-4af7-9692-33dfb9285745.png)

### Steps to Reproduce

1. Run:
$credential-Get-Credential
Invoke-ArcGISConfiguration -Mode InstallLicenseConfigure -Credential $credential
2. Wait
3. Login to AGS Server Manager and look and the registered data stores.

### Important Factoids

* The EGDB already exists as a SQL Server database and is already an enterprise geodatabase.
* I edited the [module's EGDB script](https://github.com/Esri/arcgis-powershell-dsc/blob/50a5cf49476a7059ea417dcc67f15c76e4d41522/Modules/ArcGIS/DSCResources/ArcGIS_EGDB/ArcGIS_EGDB.psm1) to  include the PowerShell Module's version # in $pythonscriptpath. Without the change, the script errored out because PowerShell couldn't find the 2 Python files. The problem may be reproducible without this workaround if you install the Esri DSC module without specifying a version # when calling Install-Module.

### References

https://github.com/Esri/arcgis-powershell-dsc/wiki/v3.2.0-Variables-reference-page-for-JSON-configuration-files
https://github.com/Esri/arcgis-powershell-dsc/blob/50a5cf49476a7059ea417dcc67f15c76e4d41522/Modules/ArcGIS/DSCResources/ArcGIS_EGDB/ArcGIS_EGDB.psm1

**Comment by cameronkroeker** [comment]
Hi @ZianAtFirstWatch,

Thanks for bringing this to our attention. We will look to get this addressed in a future release of the module, however for now here is a workaround you can use:

Update line 789:
https://github.com/Esri/arcgis-powershell-dsc/blob/35c53bca08644584440a19cb8738e7e307ecd2cf/Modules/ArcGIS/DSCResources/ArcGIS_EGDB/ArcGIS_EGDB.psm1#L789

to

```
dataStoreConnectionType = if($IsManaged){'serverOnly'}else{'shared'}
```

Thanks,
Cameron K.

**Comment by cameronkroeker** [comment]
Hi @ZianAtFirstWatch,

This issue has been addressed in the latest release, v3.3.2:

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v3.3.2

Thanks,
Cameron K.
